src/main/scala/au/com/agiledigital/toolform/tasks/generate/docker/GenerateDockerComposeV3.scala

```diff
@@ -0,0 +1,175 @@
+package au.com.agiledigital.toolform.tasks.generate.docker
+
+import java.io.{BufferedWriter, FileWriter}
+
+import au.com.agiledigital.toolform.app.{ToolFormConfiguration, ToolFormError}
+import au.com.agiledigital.toolform.model._
+import au.com.agiledigital.toolform.tasks.generate.{WriterContext, YamlWriter}
+import au.com.agiledigital.toolform.tasks.generate.docker.DockerFormatting._
+import au.com.agiledigital.toolform.tasks.generate.docker.SubEdgeDef.subEdgeDefsFromProject
+import au.com.agiledigital.toolform.version.BuildInfo
+
+import scalaz.Scalaz._
+import scalaz._
+
+class GenerateDockerComposeV3() extends YamlWriter {
+
+  // TODO: Surely Scalaz has a built in identity function that could replace this.
+  // Maybe it could be replaced with a different construct altogether?
+  private val identity = State[WriterContext, Unit] { context =>
```

Comment:
Move to companion object.

Reply:
I have moved everything in this file to a companion object because the whole thing is stateless